src/canvas/Renderer.tsx

```diff
@@ -236,15 +236,30 @@ export class Renderer implements IRenderer {
     private _endTracing = (position: Point, snapEndPoint: boolean): void => {
         if (this._currPath) {
             this._currPath.addPoint(position);
+
+            // Require a minimum of 2 points for lines and 3 points for curves
+            if ((this._toolType === ToolType.StraightLine && this._currPath.getPoints().length < 2) ||
+                (this._toolType === ToolType.Freeline && this._currPath.getPoints().length < 3)) {
+                this._discardCurrPath();
+                return;
+            } 
+            
             if (snapEndPoint) {
                 // Try to snap to other endpoints
-                const snapEndPoint = this._currPath.snapEndPoint(this._document.getPatternPaths());
+                const snappedToEndPoint = this._currPath.snapEndPoint(this._document.getPatternPaths());
+                
                 // If we were unable to snap to other endpoints, we will try to snap along other paths.
-                if (!snapEndPoint) {
+                if (!snappedToEndPoint) {
                     const snappedPosition = this._checkPointIntersectionAndSplit(position, this._document.getPatternPaths());
                     this._currPath.snapEndPointTo(snappedPosition);
                 }
             }
+            
+            // If the start is too close to the end, discard currPath
+            if (this._currPath.isFirstPointCloseToLastPoint(10)) {
```

Comment:
What if someone tries to draw a circle with the free line tool? it will disapear

Reply:
As we discussed offline. Currently you can only draw arcs, so drawing a circle and having it turn into an arc is also not ideal behavior. We will put circles in the backlog.